libp2p/pubsub/pubsub.py

```diff
@@ -361,13 +368,16 @@ def handle_subscription(
 
         # Check if this message has any topics that we are subscribed to
         for topic in publish_message.topicIDs:
-            if topic in self.my_topics:
+            if topic in self.topic_ids:
                 # we are subscribed to a topic this message was sent for,
                 # so add message to the subscription output queue
                 # for each topic
-                await self.my_topics[topic].put(publish_message)
+                await self.subscribed_topics_send[topic].send(publish_message)
 
-    async def subscribe(self, topic_id: str) -> "asyncio.Queue[rpc_pb2.Message]":
+    # TODO: Change to return an `AsyncIterable` to be I/O-agnostic?
+    async def subscribe(
+        self, topic_id: str
+    ) -> "trio.MemoryReceiveChannel[rpc_pb2.Message]":
```

Comment:
I'm thinking something like this.

```python
class SubscriptionAPI(AsyncContextManager[None], AsyncIterable[???]):
    async def __aenter__(self) -> 'SubscriptionAPI':
        ...
    async def __aexit__(self, ...):
        await self.cancel()
    async def cancel(self):
        # trigger call to `unsubscribe`
    async def __aiter__(self, ...):
        # yield results (from the trio channel)
```

This would be used like this:

```python
subscription = pubsub.subscribe(my_topic)
async for thing in subscription:
    ... 
await subscription.cancel()

# or better
async with subscription:
    async for thing in subscription:
        ...
```

This abstracts away the channel from the API.